packages/handlersjs-core/lib/daemons/scheduler.ts

```diff
@@ -0,0 +1,59 @@
+import { Observable, Subject } from 'rxjs';
+import { Daemon } from '../models/daemon';
+
+export class Scheduler extends Daemon {
+
+  private currentTimeout: NodeJS.Timeout | undefined;
+
+  constructor(
+    protected readonly interval: number,
+    protected readonly task: (() => void)
+  ) {
+
+    super();
+
+  }
+
+  start(): Observable<Daemon> {
+
+    const subject = new Subject<this>();
+
+    if (this.currentTimeout !== undefined) {
+
+      subject.error(new Error('Scheduler was already running'));
+
+    } else {
+
+      this.currentTimeout = setInterval(() => this.task(), this.interval);
```

Comment:
```suggestion
      this.currentTimeout = setInterval(this.task, this.interval);
```

Reply:
This only sets an interval for the task, not for repeating the interval (which is part of the desired functionality).